libtenzir/include/tenzir/actors.hpp

```diff
@@ -341,12 +353,15 @@ struct node_actor_traits {
       ->caf::result<std::vector<caf::actor>>,
     // Retrieve the version of the process running the NODE.
     auto(atom::get, atom::version)->caf::result<record>,
-    // Spawn a set of execution nodes for a given pipeline. Does not start the
-    // execution nodes.
-    auto(atom::spawn, operator_box, operator_type, std::string definition,
-         receiver_actor<diagnostic>, metrics_receiver_actor, int index,
-         bool is_hidden, uuid run_id)
-      ->caf::result<exec_node_actor>>
+    // Set the listening endpoint of the tenzir-node.
+    auto(atom::set, endpoint)->caf::result<void>,
+    // Spawn an pipeline_shell subprocess.
+    auto(atom::spawn, atom::shell)->caf::result<pipeline_shell_actor>,
+    // Callback from subprocess when shell actor is ready.
+    auto(atom::connect, atom::shell, uint32_t, pipeline_shell_actor)
```

Comment:
Can you name the `uint32_t` parameter?

Reply:
We don't have variable names in any of these signatures, is it really worth it to break consistency?

Reply:
We do actually name them when it's not immediately obvious. For example, look at the definition of `metrics_receiver_actor`.